pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+// Copyright 2022 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+use cfg_traits::PoolInspect;
+use codec::{Decode, Encode};
+use ethabi::{Bytes, Contract};
+use frame_support::traits::{
+	fungibles::{Inspect, Mutate, Transfer},
+	OriginTrait,
+};
+pub use pallet::*;
+use scale_info::TypeInfo;
+use sp_core::{TypeId, U256};
+use sp_runtime::{traits::AtLeast32BitUnsigned, FixedPointNumber};
+use sp_std::{boxed::Box, convert::TryInto, vec, vec::Vec};
+
+pub mod weights;
+
+mod message;
+pub use message::*;
+
+mod routers;
+pub use routers::*;
+
+/// A Domain is a chain or network we can send a Connectors message to.
+/// The domain indices need to match those used in the EVM contracts and these
+/// need to pass the Centrifuge domain to send tranche tokens from the other
+/// domain here. Therefore, DO NOT remove or move variants around.
+#[derive(Encode, Decode, Clone, PartialEq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum Domain {
+	Centrifuge,
+	Moonbeam,
+	Ethereum,
+	Avalanche,
+	Gnosis,
+}
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+pub struct DomainLocator<Domain> {
+	pub domain: Domain,
+}
+
+impl<Domain> TypeId for DomainLocator<Domain> {
+	const TYPE_ID: [u8; 4] = *b"domn";
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct DomainAddress<Domain> {
+	pub domain: Domain,
+	pub address: [u8; 32],
+}
+
+impl<Domain> TypeId for DomainAddress<Domain> {
+	const TYPE_ID: [u8; 4] = *b"dadr";
+}
+
```

Comment:
Move to common-types and ids respectively. Or do you want to keep them here for the time while connectors is still POC?

Reply:
I would keep them here while this is in POC to be honest, yeah 👍